### ISSUE TITLE
vars/cfcr.tfvars has no effect on bbl up in CFCR tutorial

### ISSUE THREAD

**Issue by awmartin** [cloudfoundry/bosh-bootloader#336]
I was following the [instructions](https://github.com/cloudfoundry/bosh-bootloader/tree/master/plan-patches/cfcr-gcp) to deploy CFCR on GCP.

The step before `bbl up` includes an instruction to create a `vars/cfcr.tfvars` file with the `kubernetes_master_host` variable thusly:

    echo kubernetes_master_host=\"${kubernetes_master_host}\" > vars/cfcr.tfvars

When I did this and ran `bbl up`, the deployment used `cfcr.your-domain.boz` as the domain, creating a DNS zone to that effect. `bbl outputs` showed this default domain as well afterwards. I would have expected to see the domain I specified instead.

I'm not sure if `bbl up` even picks up on the presence of this file. Is there a command-line arg that should have been there? I had to create another zone file and delete the default one, but now bbl doesn't match the state in GCP.

**Comment by cwlbraa** [comment]
"cfcr.your-domain.boz" is only ever mentioned in the dummy copy of that file. Is it possible you overwrote the file you made with `echo >` when you copied the patch? 

There are no additional arguments necessary. BBL should pick up any `*.tfvars` files in the vars file and pass them to terraform. I believe it does so in alphabetical order.  

```
    mkdir banana-env && cd banana-env
    bbl plan --name banana-env
    cp -r bosh-bootloader/plan-patches/cfcr-gcp/. .
    echo kubernetes_master_host=\"${kubernetes_master_host}\" > vars/cfcr.tfvars
```

the last two commands there are copying the dummy `.boz` file into your bbl state, then overwriting it with your own domain.

**Comment by awmartin** [comment]
If I understand correctly, the `echo` statement intentionally overwrites the dummy file, so I don't really understand your question. The answer would be, "Yes, of course I overwrote it with `echo`, as the instructions asked."

I know what the last two steps are _supposed_ to achieve, but they don't. The contents of my `vars/cfcr.tfvars` has my domain, but my domain wasn't deployed when I called `bbl up`. This is the behavior I believe is a bug.

**Comment by cwlbraa** [comment]
Sorry if my question was not clear in its intent. I did not mean to offend you. I'm trying to figure out where your deployment could've possibly gotten ".boz" if it is not present in "vars/cfcr.tfvars." It would've been clearer had I asked "Is it possible you overwrote the file you made with `echo >` when you `cp -r`'d the patch?"

with certain versions of that patch,
```
echo kubernetes_master_host=\"${kubernetes_master_host}\" > vars/cfcr.tfvars
cp -r bosh-bootloader/plan-patches/cfcr-gcp/. .
cat vars/cfcr.tfvars
```
will print `kubernetes_master_host="cfcr.your-domain.boz"`, just if you followed the directions slightly out of order.

If you did them in the listed order, there should be no mention of .boz domains anywhere in your bbl state or in the bbl code. Prior to responding to you initially, I had attempted to reproduce your bug but was unable to do so without switching the order to resemble the commands above. 

Perhaps try searching for the text `.boz` in your bbl state? If you find it in a tfvars file, that's where your tfvars file should be. If it's only in your tfstate, then we'll have to figure out how that happened.
